Adding support for elements that have "style" attribute

tiagorg opened the pull request:
Hi @TimBeyer 

I am trying to use your library into https://github.com/tiagorg/Marionette.VDOMView where I am trying to implement VDOM on Marionette/Backbone Views.

Your library works great, however it doesn't support elements that have "style" attribute, I found that the VirtualDOM element needs to have styles as an object rather than a string, in order for the styles to be properly materialized (created as a real DOM element from the virtual DOM).

So I would like to propose this PR (which would also bump the version) so that mine would be able to start using yours :)

Any question please let me know. Thanks!

review-ninja commented:
[![ReviewNinja](http://app.review.ninja/22070939/pull/8/badge)](http://app.review.ninja/TimBeyer/html-to-vdom/pull/8)

TimBeyer commented:
Hi @tiagorg 

Thanks a lot for the PR!
The code looks fine. You just need to add some test cases for the style attribute parsing and then I can go ahead and merge it.  
Also, no need to bump the version in the PR. I'll take care of it when I do the release.

tiagorg commented:
Absolutely! I've just added a test case for it which is expecting styles just like I need it in my lib.

I've also reverted the bumping.

Please let me know if this works.

TimBeyer commented on test/html-to-vdom/index.js:
```diff
@@ -41,6 +41,19 @@ describe('htmlparser-to-vdom', function () {
+
+        it('parses an div with styles correctly', function () {
+
+            var html = '<div style="top: -7px; left: -6px; background: rgb(0,0,132);"></input>';
```
Maybe some additional cases for missing spaces like `top:-7px;left:-6px;border:1px solid black;` might be useful.  
It also seems like browsers don't require a trailing semicolon eg: `style="top:-7px"` or `style="top: -7px; background:blue"`. I wonder if that should be supported as well.

TimBeyer commented:
I left another comment. 
It would also be super awesome if you could `git rebase -i` your branch to remove the version bump change from history completely. That way everything's nice and clean.  

http://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history
http://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

tiagorg commented:
Done and done. Please let me know if you see another deviation.

TimBeyer commented:
Just one last thing and then we're good to go:
Your tests all open a `div` tag and then close an `input` tag. They should both be divs.

tiagorg commented:
Ooops! Sorry, my mistake. Now it should be good :)

TimBeyer commented:
Released as [v0.4.0](https://github.com/TimBeyer/html-to-vdom/releases/tag/v0.4.0) 
Thanks a lot for contributing!

tiagorg commented:
Fantastic! Thanks very much Tim!
